src/index.js

```diff
@@ -1,53 +1,189 @@
+import '@fortawesome/fontawesome-free/js/all.js';
 import './style.css';
 
-// create array of objects
-const toDoList = [{
-  description: 'one',
-  completed: false,
-  index: 0,
-},
-{
-  description: 'two',
-  completed: false,
-  index: 1,
-}, {
-  description: 'three',
-  completed: false,
-  index: 2,
-}, {
-  description: 'four',
-  completed: false,
-  index: 3,
-},
-];
-// declare HTML elements
-const body = document.querySelector('body');
-// main ul 'listOfItems' class=list
-const listOfItems = document.createElement('ul');
-listOfItems.classList.add('list');
-body.appendChild(listOfItems);
-// header and input
-listOfItems.innerHTML = `
-<li class="main-header">Today's To Do<button type="submit">X</button></li>
+// Declare HTML elements
+const mainInput = document.getElementById('main-input');
+const listOfItems = document.querySelector('.list');
+const clearAll = document.querySelector('.clear-all');
 
-<li class="main-input"><input type="text" placeholder="Add to your list...">
-<button>O</button></li>
+// Create array to store items
+const itemArray = [] || localStorage.getItem('items');
+const completeItem = () => {
+  const localData = localStorage.getItem('items');
+  const parsedData = JSON.parse(localData);
+  const eachItem = document.querySelectorAll('.eachItem');
+  for (let i = 0; i < eachItem.length; i += 1) {
+    if (eachItem[i].classList.contains('strike')) {
+      parsedData[i].completed = true;
+    } else {
+      parsedData[i].completed = false;
+    }
+    localStorage.setItem('items', JSON.stringify(parsedData));
+  }
+};
+
+// Edit items
+const editItems = (oldItem) => {
+  const newInput = document.createElement('input');
+  newInput.type = 'text';
+  newInput.classList.add('new-input');
+  newInput.value = oldItem.textContent;
+  oldItem.replaceWith(newInput);
+  const local = localStorage.getItem('items');
+  const data = JSON.parse(local);
+  newInput.addEventListener('keypress', (e) => {
+    const newId = newInput.parentElement.id;
+    if (e.key === 'Enter') {
+      const oldItem = document.createElement('div');
+      oldItem.textContent = newInput.value;
+      newInput.replaceWith(oldItem);
+      const editedContainers = document.querySelectorAll('.new-item');
+      for (let i = 0; i < editedContainers.length; i += 1) {
+        if ((editedContainers[i].id) === newId) {
+          data[i].description = newInput.value;
+          localStorage.setItem('items', JSON.stringify(data));
+        }
+      }
+    }
+  });
+};
+
+// Remove items
+const removeItems = (li) => {
+  listOfItems.removeChild(li);
+  let count = 0;
+  const parsedItems = localStorage.getItem('items');
+  let localData = JSON.parse(parsedItems);
+  // Filter true elements
+  localData = localData.filter((item) => item.completed === false);
+  // Update index of elements
+  localData.map((item) => {
+    item.index = count;
+    count += 1;
+    return null;
+  });
+  localStorage.setItem('items', JSON.stringify(localData));
+};
```

Comment:
- [ ] When you remove an item from the array, the remaining items should have their indexes set to the length of the array + 1. For you to successfully implement this, I would like to advise you to initialize `count` with 1 instead of 0. 👍